iml-agent/src/action_plugins/stratagem/action_filesync.rs

```diff
@@ -5,33 +5,54 @@
 use crate::{agent_error::ImlAgentError, env, lustre::search_rootpath};
 use futures::{future::try_join_all, future::Future, stream, StreamExt, TryStreamExt};
 use iml_wire_types::{FidError, FidItem};
-use liblustreapi::LlapiFid;
 use std::path::PathBuf;
 use structopt::clap::arg_enum;
 use tokio::{fs, process::Command};
 
+arg_enum! {
+    #[derive(Debug, serde::Deserialize, Clone, Copy, PartialEq)]
+    pub enum ActionType {
+    Push,
+    Pull,
+    }
+
+}
 struct Work {
+    file_path: PathBuf,
     fid: FidItem,
-    src_file: String,
-    dest_file: String,
 }
 
 const LARGEFILE: u64 = 1024 * 1024 * 1024;
 
 fn do_rsync<'a>(
+    src_root: &'a PathBuf,
+    dest_root: &'a PathBuf,
     work_list: &'a [Work],
 ) -> impl Future<Output = Result<Vec<FidError>, ImlAgentError>> + 'a {
     stream::iter(work_list)
-        .map(|work| async move {
+        .map(move |work| async move {
+            let src_file = format!("{}/{}", src_root.display(), work.file_path.display());
+            let dest_file = format!("{}/{}", dest_root.display(), work.file_path.display());
+            let mut dest_dir = PathBuf::from(&src_file);
+            dest_dir.pop();
+            fs::create_dir_all(&dest_dir).await?;
+
             let output = Command::new("rsync")
                 .arg("-a")
                 .arg("-t")
-                .arg(&work.src_file)
-                .arg(&work.dest_file)
+                .arg(&src_file)
+                .arg(&dest_file)
                 .kill_on_drop(true)
                 .output()
                 .await?;
 
+            tracing::debug!(
+                " moved {} to {} {:?}",
+                src_file,
+                dest_file,
+                output.status.code()
+            );
```

Comment:
```suggestion
            tracing::debug!(
                " moved {} to {} {:?}",
                src_file.display(),
                dest_file.display(),
                output.status.code()
            );
```

With the above change.

Reply:
Done